### PR TITLE
Feat/refactor-email-service

### PR DESCRIPTION
**Pull request by andram11**
# Description

refactor of the email service in order to:
- create one single sendEmail function which uses variables for each type of template
- not run in test mode
- added unit tests for email service

## Issue link

Fixes # (https://app.clickup.com/t/86b2343x3)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Feature updates / changes
- [ ] Tests
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit tests were created

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have updated the change log

### CODE REVIEW

**Comment by cherylli** [comment]
@andram11 could you please resolve merge conflicts when you have a chance? Thanks!

File: src/utils/emails/email.service.ts
Comment (timDeHof) [inline src/utils/emails/email.service.ts:1]: Looks Good.

**Review by timDeHof — APPROVED** [review]
Looks great

**Review by cherylli — APPROVED** [review]
Looks good, exactly what I wanted. Looked at the code and ran all tests ✅

**Pull request merged**